docs/flask.asciidoc

```diff
@@ -0,0 +1,232 @@
+[[flask-support]]
+== Flask support
+
+Getting Elastic APM set up for your Flask project is easy,
+and there are various ways you can tweak it to fit to your needs.
+
+[float]
+[[flask-installation]]
+=== Installation
+
+Install the Elastic APM agent using pip:
+
+[source,bash]
+----
+$ pip install elastic-apm[flask]
+----
+
+or add `elastic-apm[flask]` to your project's `requirements.txt` file.
+
+NOTE: If you use Flask with uwsgi, make sure to
+http://uwsgi-docs.readthedocs.org/en/latest/Options.html#enable-threads[enable
+threads].
+
+[float]
+[[flask-setup]]
+=== Setup
+
+To set up the agent, you need to initialize it with appropriate settings.
+
+The settings are configured either via environment variables,
+the application's settings, or as initialization arguments.
+
+You can find a list of all available settings in the <<configuration, Configuration>> page.
+
+To initialize the agent for your application using environment variables:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+apm = ElasticAPM(app)
+----
+
+To configure the agent using `ELASTIC_APM` in your application's settings:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+
+app.config['ELASTIC_APM'] ={
+    'APP_NAME': '<APP-NAME>',
+    'SECRET_TOKEN': '<SECRET-TOKEN>',
+}
+apm = ElasticAPM(app)
+----
+
+The final option is to initialize the agent with the settings as arguments:
+
+[source,python]
+----
+from elasticapm.contrib.flask import ElasticAPM
+
+apm = ElasticAPM(app, app_name='<APP-ID>', secret_token='<SECRET-TOKEN>')
+----
+
+[float]
+[[flask-debug-mode]]
+==== Debug Mode
+
+Please note that errors and transactions will only be sent to the apm server
+if your app is *not* in
+http://flask.pocoo.org/docs/0.12/quickstart/#debug-mode[debug mode].
```

Comment:
💯 are you sure transactions _and_ errors? i recall transactions is fine

Reply:
That is actually quite confusing. In Django, both errors and transactions aren't recorded during debug mode. In flask, only errors aren't recorded.

I'll unify the two to both have Django's current behaviour